framework/src/bundle/BundleResourceContainer.cpp

```diff
@@ -37,15 +40,27 @@ namespace cppmicroservices {
 BundleResourceContainer::BundleResourceContainer(const std::string& location)
   : m_Location(location)
   , m_ZipArchive()
+  , m_Data()
 {
   if (!util::Exists(location)) {
     throw std::runtime_error(m_Location + " does not exist");
   }
 
-  if (!mz_zip_reader_init_file(&m_ZipArchive, m_Location.c_str(), 0)) {
-    throw std::runtime_error("Could not init zip archive for bundle at " +
-                             m_Location);
+  // Assume that the bundle had its meta-data linked into a data section.
+  // If this assumption is false, fall back to reading the meta-data in a
+  // less than optimal way, in terms of memory utilization.
+  try {
+    auto objFile = BundleObjFactory().CreateBundleFileObj(location);
+    m_Data = objFile->GetRawBundleResourceContainer();
+  } catch(...) {}
+
+  if (!m_Data ||
+      !mz_zip_reader_init_mem(&m_ZipArchive, m_Data->m_Data.get(), m_Data->m_DataSize, 0)) {
```

Comment:
`m_Data->m_Data` looks odd. Maybe `BundleResourceContainer`'s member can be something like `m_RawBundleResource`, so that this can be `m_RawBundleResource->m_Data`